src/board.cpp

```diff
@@ -0,0 +1,179 @@
+#include <iostream>
+#include "board.h"
+#include "cpu.h"
+
+
+Board::Board(size_t board_size) : players_(), num_humans_(0), board_size_(board_size) {}
+
+void Board::init(size_t new_board_size) {
+  // If board_size is 0, keep the current size
+  if (new_board_size > 0) {
+    board_size_ = new_board_size;
+  }
+  for (unsigned int i = 0; i < get_num_players(); ++i) {
+    // Player ids are 1-indexed
+    players_[i]->init(i + 1, board_size_);
+  }
+}
+
+void Board::reset() {
+  players_.clear();
+  num_humans_ = 0;
+}
+
+void Board::print_board() const {
+  GUI::print_header("NUMBERS ROYALE");
+  // Print players in reverse, so player 1 is on the bottom
+  for (int i = get_num_players() - 1; i >= 0; --i) {
+    std::string playerText = players_[i]->name() + ": " +
+        std::to_string(players_[i]->get_score()) + " points";
+    GUI::print_item(playerText.c_str());
+    GUI::print_item("Moves:");
+    GUI::print_item(players_[i]->get_moves_string().c_str(), true);
+  }
+  GUI::print_border();
+}
+
+void Board::print_scores() const {
+  int winner_id = check_game_winner();
+  switch (winner_id) {
+    case TIE_GAME:
+      GUI::print_header("TIE GAME!");
+      break;
+    case CONTINUE_GAME:
+      GUI::print_header("SCOREBOARD", false);
+      break;
+    default:
+      std::string winner_text = players_[winner_id - 1]->name() + " WINS!";
+      GUI::print_header(winner_text.c_str());
+      break;
+  }
+  // Sort players by descending score
+  std::vector<Player*> sorted_player_view;
+  for (const std::unique_ptr<Player>& player : players()) {
+      sorted_player_view.push_back(player.get()); // Add raw pointers to the view
+  }
+
+  std::sort(sorted_player_view.begin(), sorted_player_view.end(), score_cmp);
+
+  for (int i = 0; i < get_num_players(); ++i) {
+    std::string player_score_text = sorted_player_view[i]->name() + ": " +
+        std::to_string(sorted_player_view[i]->get_score()) + " points";
+    GUI::print_item(player_score_text.c_str());
+  }
+  GUI::print_border();
+}
+
+void Board::print_round_result() {
+  unsigned int winner_id = check_round_winner(); 
+
+  // If game is over, print final score and exit
+  if (check_game_winner() != CONTINUE_GAME) {
+    print_scores();
+    return;
+  }
+
+  if (winner_id == TIE_GAME) {
+    GUI::print_header("It's a tie!");
+  } else {
+    std::string winner_text = players_[winner_id - 1]->name() + " wins the round!";
+    GUI::print_header(winner_text.c_str());
+  }
+
+  // Sort players by last move, descending
+  std::vector<Player*> sorted_players_view;
+  for (const std::unique_ptr<Player>& player : players_) {
+    sorted_players_view.push_back(player.get());
+  }
+  std::sort(sorted_players_view.begin(), sorted_players_view.end(), move_cmp);
```

Comment:
const ref like above

maybe worth putting this in a helper? `get_sorted_players()`?